sparkler-api/src/main/java/edu/usc/irds/sparkler/util/FetcherDefault.java

```diff
@@ -39,6 +68,15 @@
     public FetchedData fetch(Resource resource) throws Exception {
         LOG.info("DEFAULT FETCHER {}", resource.getUrl());
         URLConnection urlConn = new URL(resource.getUrl()).openConnection();
+        if (httpHeaders != null){
+            httpHeaders.entrySet().forEach(e -> urlConn.setRequestProperty(e.getKey(), e.getValue()));
+        }
+        String userAgentValue = getUserAgent();
+        if (getUserAgent() != null) {
```

Comment:
I think we should use `userAgentValue` here. Calling `getUserAgent()` would unnecessary increment the `userAgentIndex` by 1. Please correct me if I am wrong.